src/tests/workflows/recipes/components/FilterObjectForm.test.js

```diff
@@ -0,0 +1,509 @@
+import { List, fromJS } from 'immutable';
+import { render, fireEvent } from 'react-testing-library';
+import fetchMock from 'fetch-mock';
+import { NORMANDY_ADMIN_API_ROOT_URL } from 'console/settings';
+import FilterObjectForm, {
+  serializeFilterObjectToMap,
+  deserializeFilterObjectToList,
+  InputsWidget,
+  rewriteSamplingInput,
+  SamplingInput,
+  VersionsInput,
+  parseIntOrNull,
+} from 'console/workflows/recipes/components/FilterObjectForm';
+import { createForm } from 'console/utils/forms';
+import { wrapMockStore } from 'console/tests/mockStore';
+
+describe('<FilterObjectForm>', () => {
+  it('render editing form with an empty existing filter object', () => {
+    fetchMock.getOnce(`${NORMANDY_ADMIN_API_ROOT_URL}v3/filters/`, {
+      channels: [
+        {
+          key: 'beta',
+          value: 'Beta',
+        },
+      ],
+      countries: [{ key: 'SE', value: 'Sweden' }],
+      locales: [{ key: 'sv', value: 'Swedish' }],
+    });
+
+    const filterObject = serializeFilterObjectToMap(fromJS([]));
+    const props = {
+      disabled: false,
+      form: {
+        getFieldValue: key => {
+          // Remember, 'key' is something like "filter_object.locales"
+          const value = filterObject.get(key.split('.')[1]);
+          if (value) {
+            return value.toJS();
+          }
+        },
+        getFieldDecorator: (key, config) => {
+          return Component => {
+            return;
+          };
+        },
+      },
+      filterObject,
+      onSubmit: jest.fn(),
+    };
+
+    const FakeForm = createForm({})(FilterObjectForm);
+    const { getByText } = render(wrapMockStore(<FakeForm {...props} />));
+    expect(getByText('Geo')).toBeTruthy();
+    expect(getByText('Browser')).toBeTruthy();
+    expect(getByText('Sampling')).toBeTruthy();
+    expect(fetchMock.called()).toBeTruthy();
```

Comment:
There is a lot going on in this test just to render an empty form. Is it really all needed?

Reply:
As a matter of fact I had a bug in the early days where I had written code that worked for editing recipes but threw an error when I tried to add a recipe. Adding a recipe is like editing with an empty filter_object.